app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -2,10 +2,29 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   include PreferencesController
 
   require_authentication :all, scopes: [:project]
+  resource_actions :create, :update, :show, :index, :update_settings
+  extra_schema_actions :update_settings
   schema_type :json_schema
+  before_action :validate_records, only: [:update_settings]
+
+  def update_settings
+    upp = UserProjectPreference.where(project_id: params_for[:project_id],
+                                      user_id: params_for[:user_id]).first
+    upp.settings.merge! params_for[:settings]
+    upp.save!
+    render status: :ok, nothing: true
+  end
 
   private
 
+  def validate_records
+    User.find params_for[:user_id]
```

Comment:
Call to User was just to confirm its existence and raise an AR exception if it didn't. Seems like I can have it both ways and save @upp in the validation before_action and not have to make more than the one call, since I'm not potentially initializing a UPP anymore.